pkg/engine/planner/logical/planner.go

```diff
@@ -102,9 +131,9 @@ func convertLabelMatchers(matchers []*labels.Matcher) Value {
 func convertMatcherType(t labels.MatchType) types.BinaryOp {
 	switch t {
 	case labels.MatchEqual:
-		return types.BinaryOpEq
+		return types.BinaryOpMatchStr
```

Comment:
This seems like we have two equality binary operations: one for strings and one for everything else. Do you think we need both?

Reply:
Yes, I think we should keep both. `BinaryOpEq` is an exact equality comparison, where in a filter, the `BinaryOpMatchStr` is a contains comparison.

That said, we should use `BinaryOpEq` and `BinaryOpNeq` for the label matchers then.

Reply:
Thanks, I think I was thrown off by how "match" reads like it's also an equality check. Should we change it to `BinaryOpStrContains` to be explicit?